chainer/training/extensions/_snapshot.py

```diff
@@ -81,12 +81,14 @@ def snapshot(trainer):
 def _snapshot_object(trainer, target, filename, savefun):
     fn = filename.format(trainer)
     prefix = 'tmp' + fn
-    fd, tmppath = tempfile.mkstemp(prefix=prefix, dir=trainer.out)
+
+    tmpdir = tempfile.TemporaryDirectory(dir=trainer.out)
```

Comment:
Please use `prefix` here.

Reply:
`TemporaryDirectory` seems unavailable in Python 2.7.

Reply:
Good catch. Please use `tempfile.mkdtemp` instead.